roctracer changes to support multiple ROCm installation

pruthvistony opened the pull request:
roctracer changes to support multiple ROCM installation

- Package is generated to install into ROCM_PATH by setting
      to CMAKE_INSTALL_PREFIX, if defined in the env otherwise
      default into /opt/rocm
- Lib SO version is added dependent on build version
- RUNPATH is set to a default value based on /opt/rocm and if
      ROCM_RPATH env is defined it is overwritten.
- Symlinks are created for library so files.
Signed-off-by: Pruthvi Madugundu <mpruthvi@gmail.com>

eshcherb commented on build.sh:
```diff
@@ -1,7 +1,10 @@
+ROCM_STACK_PATH="/opt/rocm"
+if ! [ -z ${ROCM_PATH+x} ]; then
+    ROCM_STACK_PATH=${ROCM_PATH}
+fi
```
Could you change on:
ROCM_PATH=${ROCM_PATH:-/opt/rocm}

And you don't need to rename it to ROCM_STACK_PATH

pruthvistony replied:
Agree. I have updated the code accordingly.

eshcherb commented on CMakeLists.txt:
```diff
@@ -43,17 +43,27 @@ include ( env )
 set ( BUILD_VERSION_MAJOR ${VERSION_MAJOR} )
 set ( BUILD_VERSION_MINOR ${VERSION_MINOR} )
 set ( BUILD_VERSION_PATCH ${VERSION_PATCH} )
-set ( LIB_VERSION_STRING "${BUILD_VERSION_MAJOR}.${BUILD_VERSION_MINOR}.${BUILD_VERSION_PATCH}" )
```
It seems that BUILD_VERSION_* is not used more, so why areyou keeping them?

pruthvistony replied:
I have updated the code, please check the latest commit.

eshcherb commented on CMakeLists.txt:
```diff
@@ -43,17 +43,27 @@ include ( env )
   message ( "VERSION BUILD DEFINED ${VERSION_BUILD}" )
   set ( BUILD_VERSION_PATCH "${BUILD_VERSION_PATCH}-${VERSION_BUILD}" )
 endif ()
 set ( BUILD_VERSION_STRING "${BUILD_VERSION_MAJOR}.${BUILD_VERSION_MINOR}.${BUILD_VERSION_PATCH}" )
```
It seems that BUILD_VERSION_STRING is not used also, just in the message below.

pruthvistony replied:
I have updated the code, please check the latest commit.

eshcherb commented on CMakeLists.txt:
```diff
@@ -43,17 +43,27 @@ include ( env )
   set ( BUILD_VERSION_PATCH "${BUILD_VERSION_PATCH}-${VERSION_BUILD}" )
 endif ()
 set ( BUILD_VERSION_STRING "${BUILD_VERSION_MAJOR}.${BUILD_VERSION_MINOR}.${BUILD_VERSION_PATCH}" )
+message (STATUS "BUILD-VERSION: ${BUILD_VERSION_STRING}" )
```
What the meaning of this message?

pruthvistony replied:
This message is removed. I have updated the code, please check the latest commit.

eshcherb commented on CMakeLists.txt:
```diff
@@ -43,17 +43,27 @@ include ( env )
+  set (LIB_VERSION_PATCH 0)
+endif()
+set ( LIB_VERSION_STRING "${LIB_VERSION_MAJOR}.${LIB_VERSION_MINOR}.${LIB_VERSION_PATCH}" )
+message ( "-- LIB-VERSION: ${LIB_VERSION_MAJOR}.${LIB_VERSION_MINOR}.${LIB_VERSION_PATCH}" )
```
So the MAJOR and MINOR are defined statically?
And how ROCM_LIBPATCH_VERSION is supposed to be defined?

pruthvistony replied:
Major and Minor numbers of the lib are made same BUILD_MAJOR and BUILD_MINOR. 
ROCM_LIBPATCH_VERSION is defined in the build env of jenkins and if not defined it defaults to BUILD_PATCH.
Please check the updated commit.

pruthvistony commented:
Please check the latest commit.
- I have added multiple so version symlinks similar to other libs followed in ROCm.
- Added the so versioning to libroctx64.so, as part of the initial changes as it was missed for this lib.

pruthvistony commented:
Please review the latest commit.

pruthvistony commented:
Please review the latest commit.

eshcherb commented on build.sh:
```diff
@@ -1,7 +1,8 @@
 COMPONENT="roctracer"
-ROCM_PATH="/opt/rocm"
+ROCM_PATH="${ROCM_PATH:=/opt/rocm}"
+LD_RUNPATH_FLAG=" -Wl,--enable-new-dtags -Wl,--rpath,/opt/rocm/lib:/opt/rocm/lib64"
```
Could you change it as:
LD_RUNPATH_FLAG=" -Wl,--enable-new-dtags -Wl,--rpath,$ROCM_PATH/lib:$ROCM_PATH/lib64"

And you don;'t need in the line 23

pruthvistony replied:
Ok. I will make this change.

eshcherb commented on build.sh:
```diff
@@ -19,6 +20,7 @@ if [ -z "$PACKAGE_ROOT" ] ; then PACKAGE_ROOT=$ROCM_PATH; fi
 if [ -z "$PACKAGE_PREFIX" ] ; then PACKAGE_PREFIX="$ROCM_PATH/$COMPONENT"; fi
 if [ -z "$PREFIX_PATH" ] ; then PREFIX_PATH=$PACKAGE_ROOT; fi
 if [ -n "$HIP_VDI" ] ; then HIP_VDI_OPT="-DHIP_VDI=1"; fi
+if ! [ -z ${ROCM_RPATH+x} ] ; then LD_RUNPATH_FLAG=" -Wl,--enable-new-dtags -Wl,--rpath,${ROCM_RPATH}"; fi
```
There is an error. It should have "--rpath,${ROCM_RPATH}/lib".
And you don't need this line at all, so please remove it.

pruthvistony replied:
ROCM_RPATH is different from ROCM_PATH. So this line is required, where we are setting the RUNPATH through LD FLags.

eshcherb replied:
Why use this 'if ! [ -z ${ROCM_RPATH+x} ]', just 'if [ -n "ROCM_RPATH" ]'?

pruthvistony replied:
Both would work, ${ROCM_RPATH+x} used parameter expansion which evaluates to nothing if ROCM_RPATH is unset. I see that the PR is merged, so let me know If I need to raise a new PR.

pruthvistony replied:
The above changes have been done already. Thank you for merging the patch.

eshcherb replied:
Thank you!
